go/multiorch/recovery/analysis/shard_needs_bootstrap_analyzer.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analysis
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/multigres/multigres/go/multiorch/store"
+	clustermetadatapb "github.com/multigres/multigres/go/pb/clustermetadata"
+)
+
+// ShardNeedsBootstrapAnalyzer detects when all nodes in a shard are uninitialized.
+// This is a per-pooler analyzer that returns a shard-wide problem.
+// The recovery loop's filterAndPrioritize() will deduplicate multiple instances.
+type ShardNeedsBootstrapAnalyzer struct{}
+
+func (a *ShardNeedsBootstrapAnalyzer) Name() CheckName {
+	return "ShardNeedsBootstrap"
+}
+
+func (a *ShardNeedsBootstrapAnalyzer) Analyze(poolerAnalysis *store.ReplicationAnalysis) []Problem {
+	// Skip unreachable nodes - we can't determine their true initialization state.
+	// An unreachable node might be perfectly initialized but just temporarily down.
+	// Other analyzers (PrimaryIsDead, ShardHasNoPrimary) will handle dead primaries.
+	if poolerAnalysis.IsUnreachable {
+		return nil
+	}
+
+	// Skip primary nodes - they don't have a PrimaryPoolerID by design (they ARE the primary).
+	// A dead primary should be handled by PrimaryIsDead (detected by replicas), not by
+	// ShardNeedsBootstrap. When a primary's postgres dies, it appears as:
+	// - IsPrimary = true (it's the primary)
+	// - IsInitialized = false (postgres down)
+	// - PrimaryPoolerID = nil (it's the primary itself)
+	// This would incorrectly trigger ShardNeedsBootstrap if we don't skip it.
+	// Note: IsPrimary is based on MultiPooler.Type == PRIMARY from topology, which is set
+	// when the node is initialized as primary. New uninitialized nodes start with Type=PRIMARY
```

Comment:
Need to fix this comment. Tests and provisioner were previously starting all poolers as PRIMARY, but I have fixed that.